### PR TITLE
drivers: modem: add Telit ME910G1

### PR DESCRIPTION
**Pull request by jeffwelder-ellenbytech**
Added Telit ME910G1 dts and modem chat scripts.

Refactor device macro to be custom for each modem to allow custom configurations on each modem (assisted by @bjarki-trackunit). 
(This was needed by Telit ME910G1 to allow for ON_OFF to be toggled for minimum of 5 seconds as per Telit)

### CODE REVIEW

**Comment by bjarki-trackunit** [comment]
Looks good :) FYI it will take a while to get merged because of the feature freeze :)

**Review by bjarki-andreasen — DISMISSED** [review]
Nice work :)

**Comment by bjarki-trackunit** [comment]
@rerickson1 Could you review this :)

**Review by ycsin — COMMENTED** [review]
From the looks of it, it seems that `modem_cellular.c` can grow indefinitely, as the number of supported modem grows?

File: dts/bindings/modem/telit,me910g1.yaml
Comment (fabiobaltieri) [inline dts/bindings/modem/telit,me910g1.yaml:2]: This probably wants to be `SPDX-License-Identifier: Apache-2.0`

**Comment by bjarki-trackunit** [comment]
@ycsin The driver is capable of supporing an "infinite" amount of modems, as long as said modems implement the standard AT, CMUX and PPP. However, the driver is an MVP, so it will not get any fancy non generic features, like setting radio access tech, sending sms, internal gnss, etc. so it should not grow in complexity :)

File: dts/bindings/modem/telit,me910g1.yaml
Comment (rerickson1) [reply dts/bindings/modem/telit,me910g1.yaml:2]: @jeffwelder-ellenbytech please fix this.

**Comment by jeffwelder-ellenbytech** [comment]
Fixed the licensing. 
Working on an improved chat script that works with various eSIMs before merge, will commit again once that is complete.

**Review by ycsin — DISMISSED** [review]

**Comment by bjarki-trackunit** [comment]
> Fixed the licensing. Working on an improved chat script that works with various eSIMs before merge, will commit again once that is complete.

Love it! I have added the DNM (Do Not Merge) label until you merge the fixes, you can remove it and re-request reviews when ready :)

**Comment by jeffwelder-ellenbytech** [comment]
I updated the chat scripts to have a more stable experience with the modem. Specifically I was having trouble with Verizon/AT&T eSIMs either giving an error or taking unusually long for setting the APN. 

**Init Script:**
- Added PDP Context Setting (Allows retry on error by restarting init script)

**Dial Script:**
- Remove PDP Context Activation (Note: second parameter not supported by ME910G1 by current firmware)
- Remove PDP Context Setting (Done in init script to allow error retries)
- Added Dummy AT command (Dial script doesn't work if it is only 1 command long)

If the network is too fast to register, the state machine in the dial script will miss the unsolicited registration status, and hang forever. This is fixed in @bjarki-trackunit 's #64659 by polling the registration state which I tested this morning as working. 😄

**Review by rerickson1 — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**
